### PR TITLE
Add S3 download script

### PR DESCRIPTION
**Pull request by hachibu**
Add S3 download script.

### CODE REVIEW

File: s3/download.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+	Workers int
+}
+
+func saveFile(file *s3.GetObjectOutput, filePath string) error {
+	out, err := os.Create(filePath)
+	if err != nil {
+		return err
+	}
+	io.Copy(out, file.Body)
+	if err != nil {
```
Comment (marinhero) [inline s3/download.go]: no assignation is performed after line 45, why check for error again?

File: s3/download.go [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+	"time"
+
+	"../benchmark"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+)
+
+func check(err error) {
```
Comment (marinhero) [inline s3/download.go]: checkAndExit?

Comment (hachibu) [inline s3/download.go]: What about `eoe` (Exit On Error) to be really Go and cryptic?

```go
func eoe(err error) {}
```

File: s3/download.go (hunk 1 above)
Comment (hachibu) [inline s3/download.go]: Oh yeah, I meant to check the err from Copy. I'll fix this. It should be:
```go
_, err = io.Copy(out, file.Body)
```

File: s3/download.go (hunk 2 above)
Comment (marinhero) [inline s3/download.go]: I like the name, let's keep it without contraction,  just: 
```go
func exitOnErr(err error) {...}
```

Comment (hachibu) [inline s3/download.go]: 👍

Comment (hachibu) [inline s3/download.go]: But look how cool this looks!
```go
file, err := config.Service.GetObject(objectInput)
eoe(err)
```

Comment (hachibu) [inline s3/download.go]: @marinhero I went with `exitOnError`

**Comment by hachibu** [comment]
@marinhero I'm going to add argument parsing before I merge this in.

**Pull request merged**